cmd/operator-sdk/generate/bundle/bundle.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+
+	genutil "github.com/operator-framework/operator-sdk/cmd/operator-sdk/generate/internal"
+	gencsv "github.com/operator-framework/operator-sdk/internal/generate/clusterserviceversion"
+	"github.com/operator-framework/operator-sdk/internal/generate/collector"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/kustomize"
+)
+
+// TODO(estroz): make these paths relative to inputDir
+const kubebuilderKustomization = `resources:
+- ../default
+- ../samples
+`
+
+func (c *bundleCmd) setDefaults(cfg *config.Config) {
+	if c.operatorName == "" {
+		c.operatorName = filepath.Base(cfg.Repo)
+	}
+}
+
+func (c bundleCmd) runKustomize(cfg *config.Config) error {
+
+	if !c.quiet {
+		fmt.Println("Generating bundle manifest kustomize bases")
+	}
+
+	defaultDir := filepath.Join("config", "bundle")
+	if c.inputDir == "" {
+		c.inputDir = defaultDir
+	}
+	if c.outputDir == "" {
+		c.outputDir = defaultDir
+	}
+	if c.apisDir == "" {
+		if cfg.MultiGroup {
+			c.apisDir = "apis"
+		} else {
+			c.apisDir = "api"
+		}
+	}
+
+	csvGen := gencsv.Generator{
+		OperatorName: c.operatorName,
+		OperatorType: genutil.PluginKeyToOperatorType(cfg.Layout),
+	}
+	opts := []gencsv.Option{
+		gencsv.WithBase(c.inputDir, c.apisDir),
+		gencsv.WithBaseWriter(c.outputDir),
+	}
+	if err := csvGen.Generate(cfg, opts...); err != nil {
+		return fmt.Errorf("error generating ClusterServiceVersion: %v", err)
+	}
+
+	if err := kustomize.WriteIfNotExist(c.outputDir, kubebuilderKustomization); err != nil {
```

Comment:
Are we not here addding bundle metafest files to allow customizations?
```suggestion
	if err := kustomize.WriteIfNotExist(c.outputDir, bundleKustomization); err != nil {
```

Reply:
No, `make bundle` runs `kustomize build` on those directories so any changes (ex. patches) a user wants applied to their manifests occur before being read by `generate bundle`.

Reply:
May is valid check if the code can be more intuitive otherwise add comments to clarifies. Note that, my concern here is more about manutence ability. I mean,  be sure that any person that gets this code is able to understand what/where/why. e.g address in the code common/possible questions that will make for who tries to do a fix or rfe in this area: what diff has the kustomize implemented here with the default make one? Are we creating new files specifics for the bundle or just using the preexistent ones?